wagtail_footnotes/blocks.py

```diff
@@ -29,7 +30,7 @@ def replace_footnote_tags(self, value, html, context=None):
         else:
             new_context = self.get_context(value, parent_context=dict(context))
 
-        if "page" not in new_context:
+        if "page" not in new_context or not isinstance(new_context["page"], Page):
```

Comment:
why not:

```suggestion
        if not isinstance(new_context.get("page"), Page):
```

if "page" is not in new_context, `new_context.get("page")` is `None` so the `isinstance()` check will be false.